src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -25,30 +24,73 @@ export const MultiSelectDialog = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectDialogProps>>(
     (props, ref) => {
       const {
+        helperText,
         id,
         isBlockElement,
         isDefaultOpen,
+        isSearchable,
+        defaultItemsVisible,
         items,
-        labelText,
+        listOverflow,
+        buttonText,
+        closeOnBlur,
         onApply,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
+        showLabel,
+        showAll,
         width,
         ...rest
       } = props;
 
-      // Use NYPL Breakpoints hook to check screen size
-      const { isLargerThanMobile } = useNYPLBreakpoints();
-
       // Control the open or closed state of the MultiSelect.
-      const [isOpen, setIsOpen] = useState(isDefaultOpen);
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+      const [viewAllLabel, setViewAllLabel] = useState("");
+      const [listHeight, setListHeight] = useState("");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+      const [defaultCheckboxes, setDefaultCheckboxes] = useState("none");
+      const [defaultChildCheckboxes, setDefaultChildCheckboxes] =
+        useState("none");
+      //const [prevIsOpen, setPrevIsOpen] = React.useState(isDropDownOpen);
+
+      React.useEffect(() => {
+        const calculateListHeight = () => {
+          if (listOverflow === "scroll") {
+            setViewAllLabel("View all");
+            setListHeight("260px");
+            setListItemsCount(defaultItemsVisible);
+          } else {
+            setViewAllLabel("View less");
+            setListHeight("");
+            setListItemsCount(items.length);
+          }
+        };
+
+        calculateListHeight();
+
+        if (listItemsCount === defaultItemsVisible) {
+          displayDefaultItems();
+        } else {
+          setItemsList(items);
+        }
+      }, []);
```

Comment:
When you use `useEffect` with an empty dependency array, then you are only calling it once when it renders. I'm not sure if this is what you want. If it is, then you probably don't need the `useEffect` and need to refactor your code. Otherwise, you need to add the variables that are used inside the `useEffect` into this array.

This is a screenshot of the linter giving a warning:
<img width="1125" alt="Screen Shot 2023-12-12 at 12 14 27 PM" src="https://github.com/NYPL/nypl-design-system/assets/1280564/c39ca5b4-d8e1-465a-8b31-494fbc0e2f37">